types/fh-sync.d.ts

```diff
@@ -0,0 +1,121 @@
+declare module 'fh-sync' {
+
+  interface PendingWorkerBackoff {
+    strategy: string;
+    max: string;
+  }
+
+  interface AckWorkerBackoff {
+    strategy: string;
+    max: string;
+  }
+
+  interface SyncWorkerBackoff {
+    strategy: string;
+    max: number;
+  }
+
+  interface SyncGlobalOptions {
+    pendingWorkerInterval?: number;
+    pendingWorkerConcurrency?: number;
+    pendingWorkerBackoff?: PendingWorkerBackoff;
+    ackWorkerInterval?: number;
+    ackWorkerConcurrency?: number;
+    ackWorkerBackoff?: AckWorkerBackoff;
+    syncWorkerInterval?: number;
+    syncWorkgerConcurrency?: number;
+    syncWorkerBackoff?: SyncWorkerBackoff;
+    schedulerInterval?: number;
+    schedulerLockMaxTime?: number;
+    schedulerLockName?: string;
+    datasetClientUpdateConcurrency?: number;
+    collectStats?: boolean;
+    statsRecordsToKeep?: number;
+    collectStatsInterval?: number;
+    metricsInfluxdbHost?: string;
+    metricsInfluxdbPort?: number;
+    metricsReportConcurrency?: number;
+    useCache?: boolean;
+    queueMessagesTTL?: string;
+    datasetClientCleanerCheckFrequency?: string;
+  }
+
+  type StandardCb<T> = (err: Error | string | undefined, res: T | undefined) => void;
+  type NoRespCb = (err: Error | string | undefined) => void;
+
+  interface SyncInitOptions {
+    /**
+     * Value indicating how often the dataset client should be sync with the backend. Matches the clients default 
+     * frequency. Value in seconds
+     */
+    syncFrequency?: number,
+
+    /**
+     * Value that will be used to decide if the dataset client is not active anymore.
+     */
+    clientSyncTimeout?: number,
+
+    /**
+     * Value that determines how long it should wait for the backend list operation to complete
+     */
+    backendListTimeout?: number,
+
+    /**
+    * Specify the max wait time the dataset can be scheduled to sync again after its previous schedule, in seconds.
+    */
+    maxScheduleWaitTime?: number
+  }
+
+  interface SyncInterceptParams {
+    query_params: any;
+    meta_data: any;
+  }
+
+  namespace Sync {
+    function connect(mongoDBConnectionUrl: string, mongoDBConnectionOption: any, redisUrl: string, cb: StandardCb<void>)
+
+    function init(dataset_id: string, options: SyncInitOptions, callback: StandardCb<void>): void;
+    function invoke(dataset_id: string, options: any, callback: () => void): void;
+
+    function stop(dataset_id: string, onStop: () => void): void;
+    function stopAll(onstop: StandardCb<string[]>): void;
```

Comment:
Small thing: `onStop`. Also, I don't see any reason why the snakecase variables here can't be renamed in the declaration file `datasetId` etc. WDYT?

I think that's the last thing so I'll give it a go in a project if no one else is verifying?